crates/router/src/core/refunds_v2.rs

```diff
@@ -0,0 +1,798 @@
+                            updated_by: storage_scheme.to_string(),
+                            connector_refund_id: None,
+                            processor_refund_data: None,
+                            unified_code: None,
+                            unified_message: None,
+                        })
+                    }
+                    _ => None,
+                });
+        // Update the refund status as failure if connector_error is NotImplemented
+        if let Some(refund_error_update) = option_refund_error_update {
+            state
+                .store
+                .update_refund(
+                    refund.to_owned(),
+                    refund_error_update,
+                    merchant_account.storage_scheme,
+                )
+                .await
+                .to_not_found_response(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable_lazy(|| {
+                    format!(
+                        "Failed while updating refund: refund_id: {}",
+                        refund.id.get_string_repr()
+                    )
+                })?;
+        }
+        let mut refund_router_data_res = router_data_res.to_refund_failed_response()?;
+        // Initiating Integrity check
+        let integrity_result = check_refund_integrity(
+            &refund_router_data_res.request,
+            &refund_router_data_res.response,
+        );
+        refund_router_data_res.integrity_check = integrity_result;
+        refund_router_data_res
+    } else {
+        router_data
+    };
+
+    let refund_update = match router_data_res.response {
+        Err(err) => {
+            let option_gsm = helpers::get_gsm_record(
+                state,
+                Some(err.code.clone()),
+                Some(err.message.clone()),
+                connector.connector_name.to_string(),
+                consts::REFUND_FLOW_STR.to_string(),
+            )
+            .await;
+            // Note: Some connectors do not have a separate list of refund errors
+            // In such cases, the error codes and messages are stored under "Authorize" flow in GSM table
+            // So we will have to fetch the GSM using Authorize flow in case GSM is not found using "refund_flow"
+            let option_gsm = if option_gsm.is_none() {
+                helpers::get_gsm_record(
+                    state,
+                    Some(err.code.clone()),
+                    Some(err.message.clone()),
+                    connector.connector_name.to_string(),
+                    consts::AUTHORIZE_FLOW_STR.to_string(),
+                )
+                .await
+            } else {
+                option_gsm
+            };
+
+            let gsm_unified_code = option_gsm.as_ref().and_then(|gsm| gsm.unified_code.clone());
+            let gsm_unified_message = option_gsm.and_then(|gsm| gsm.unified_message);
+
+            let (unified_code, unified_message) = if let Some((code, message)) =
```

Comment:
Use match instead if let some for options and try to write this in idiomatic way